CliWrap/Utils/WindowsSignaler.cs

```diff
@@ -46,6 +46,12 @@ public void Dispose()
         try
         {
             File.Delete(filePath);
+
+            // On Windows calling GetTempFileName creates an 0 size unique tmp*.tmp file to ensure the availability of the generated temp file name
+            // We need to delete this ourselves to avoid tmp file leaking (The generated temp files are not automatically deleted)
+            var tempPath = Path.ChangeExtension(filePath, "tmp");
+            if (File.Exists(tempPath))
+                File.Delete(tempPath);
```

Comment:
What if instead of doing this, we just avoid using `Path.GetTempFileName()` and instead do `Path.Combine(Path.GetTempPath(), "generate some semi-random string.exe")`?

Reply:
Good idea :-) I actually do this my self, using GetTempPath() + UUID to generate collision free unique temp filenames; I don't the think the old 8.3 filename restriction from DOS days applies anymore, for any version of Windows supporting .NET Standard 2.0